### PR TITLE
done HW3

### PR DESCRIPTION
**Pull request by LokrionMe**

### CODE REVIEW

File: HW3/bag.py
Comment (samodurOFF) [inline HW3/bag.py:1]: Принято. Но если хотите усложнить себе задачу и занять вечер размышлениями, то вот вам дополнительно условие: вернуть все комбинации укладки рюкзака, при которых добавление хотя бы еще одной вещи приведет к его переполнению.

File: HW3/big_text.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+from string import punctuation
```
Comment (samodurOFF) [inline HW3/big_text.py:1]: Хорошая идея. Можно еще использовать метод строк `isalpha` при переборе символов текста, пропуская кроме букв еще и пробел.

File: HW3/big_text.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+b_text = "Python also includes a data type for sets. A set is an unordered collection with"\
+    " no duplicate elements. Basic uses include membership testing and eliminating duplicate"\
+    " entries. Set objects also support mathematical operations like union, intersection,"\
+    " difference, and symmetric difference.\n"\
+    "Curly braces or the set() function can be used to create sets. Note: to create an empty"\
+    "set you have to use set(), not {}; the latter creates an empty dictionary, a data "\
+    "structure that we discuss in the next section."
+list_punct = []
+for i in range(len(punctuation)):
+    list_punct.append(punctuation[i])
```
Comment (samodurOFF) [inline HW3/big_text.py:11]: Во-первых, зачем вам переводить строку с пунктуацией в список пунктуации? В цикле в 12 строке можно было проходиться напрямую по строке `punctuation`. Во-вторых, зачем это делать таким образом, когда просто можно распаковать строку в литерал.
```suggestion
list_punct = [*punctuation]
```

File: HW3/big_text.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    for key, val in outter_lib.items():
+        if val == maximum:
+            count_space = len_max_word + len_max_numb + 3 - \
+                len(str(point)) - len(key)
+            print(f"{point}. {key}{val:>{count_space}}")
+            k -= 1
+            point += 1
+        if k == 0:
+            break
+    maximum -= 1
```
Comment (samodurOFF) [inline HW3/big_text.py:41]: Вообще всего этого можно было не делать, а просто передавать функции `sorted` сразу же созданный список слов `list_words`. Нужно просто правильно сформировать аргумент `key`.

File: HW3/friends.py
Comment (samodurOFF) [inline HW3/friends.py:1]: Проверял только вывод. Он верный, добавить нечего.

File: HW3/list_double.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+
+def find_double(inner_list: list) -> list:
+    outer_list = []
+    for item in set(inner_list):
+        if inner_list.count(item)>1:
```
Comment (samodurOFF) [inline HW3/list_double.py:5]:
```suggestion
        if inner_list.count(item) > 1:
```

File: HW3/list_double.py
Comment (samodurOFF) [inline HW3/list_double.py:1]: Верно
